include/oneapi/dpl/pstl/utils.h

```diff
@@ -287,6 +287,42 @@ class __transform_functor
     }
 };
 
+template <typename __UnaryOper, typename __UnaryPred>
+class __transform_if_unary_functor
+{
+    mutable __UnaryOper _M_oper;
+    mutable __UnaryPred _M_pred;
+
+  public:
+    explicit __transform_if_unary_functor(__UnaryOper __op, __UnaryPred __pred) : _M_oper(__op), _M_pred(__pred) {}
```

Comment:
In the constructor initializer list, it might be better to do `_M_oper(::std::move(__op)), _M_pred(::std::move(__pred))` to prevent an additional copies.

Reply:
I'm following the convention of most of the other functors in this file when dealing with predicates / opers, which do not make use of `std::move()`.  If we want to make this change, we should consider applying it to the other functors as well.

Reply:
I made this change, it is for the better.  Thanks.

Reply:
I believe this semantic assume that in any case we have one copy of this param on caller side.
May be better to use universal reference ```&&``` and ```std::forward``` ?

Reply:
Yes, this is a further improvement.  I've added this.